src/storage/src/source/generator/key_value.rs

```diff
@@ -0,0 +1,584 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+use std::convert::Infallible;
+
+use differential_dataflow::{AsCollection, Collection};
+use futures::stream::StreamExt;
+use mz_ore::cast::CastFrom;
+use mz_repr::{Datum, Diff, Row};
+use mz_storage_types::sources::load_generator::KeyValueLoadGenerator;
+use mz_storage_types::sources::{MzOffset, SourceTimestamp};
+use mz_timely_util::builder_async::{OperatorBuilder as AsyncOperatorBuilder, PressOnDropButton};
+use rand::rngs::StdRng;
+use rand::{RngCore, SeedableRng};
+use timely::dataflow::operators::{Concat, ToStream};
+use timely::dataflow::{Scope, Stream};
+use timely::progress::Antichain;
+
+use crate::healthcheck::{HealthStatusMessage, HealthStatusUpdate, StatusNamespace};
+use crate::source::types::ProgressStatisticsUpdate;
+use crate::source::{RawSourceCreationConfig, SourceMessage, SourceReaderError};
+
+pub fn render<G: Scope<Timestamp = MzOffset>>(
+    key_value: KeyValueLoadGenerator,
+    scope: &mut G,
+    config: RawSourceCreationConfig,
+    resume_uppers: impl futures::Stream<Item = Antichain<MzOffset>> + 'static,
```

Comment:
Let's start using this new name
```suggestion
    committed_upper: impl futures::Stream<Item = Antichain<MzOffset>> + 'static,
```